sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClient.java

```diff
@@ -298,11 +297,21 @@ public void run() {
         return new ServiceCall() {
 
             @Override
-            public void cancel() {
-                if (!call.isCancelled()) {
-                    call.cancel(true);
+            public boolean ensureFinished() {
+                try {
+                    return call.isHttpResult(call.get());
+                } catch (Exception ignored) {
+                    return false;
                 }
             }
+
+            @Override
+            public void cancel() {
+
+                // FIXME This doesn't kill the AsyncTask!
```

Comment:
Yeah and basically we shouldn't cancel on network outage, let call fail, and if it somehow succeeds, accept callback, we don't need all these changes, we should instead avoid cancelling anything from network state decorator if already running, and avoid resuming call if still running.

I don't understand this PR at all overall and should use a different approach of dealing with network state but not in the way you modified so far. Let call run, just pause new calls, forward success callback to channel, let channel accept it.

Reply:
For the cancel part, I would keep the code as before and cancel would be kept only for disabling scenario (where we delete logs), but as said on previous comment, don't call cancel on network down.